tutorials/tutorial-forecasting.md

```diff
@@ -0,0 +1,773 @@
+function into `fit` and get a summary of the ARIMA model that has been created. 
+
+```r
+# Install and load the forecast package needed for ARIMA
+install.packages("forecast")
+library("forecast")
+
+# use auto.arima to automatically get the arima model parameters with best fit
+fit <- auto.arima(xts_count_rides[,1], D = 1, seasonal = TRUE)
+
+# see the summary of the fit
+summary(fit)
+Series: xts_count_rides[, 1]
+ARIMA(4,0,2)(0,1,0)[168] with drift
+
+Coefficients:
+         ar1      ar2     ar3     ar4      ma1     ma2   drift
+      2.3211  -1.8758  0.3959  0.1001  -1.7643  0.9444  0.3561
+s.e.  0.0634   0.1487  0.1460  0.0588   0.0361  0.0307  0.0705
+
+sigma^2 estimated as 5193:  log likelihood=-1911.21
+AIC=3838.42   AICc=3838.86   BIC=3868.95
+
+Training set error measures:
+                     ME     RMSE      MAE       MPE     MAPE      MASE
+Training set -0.2800571 58.22306 33.15943 -1.783649 7.868031 0.4257707
+                    ACF1
+Training set -0.02641353
+```
+
+Finally, the ARIMA model can be used to forecast future values. 
+The `h` parameter specifies the number of steps we want to forecast. 
+
+```r
+# forecast future values using the arima model, h specifies the number of readings to forecast
+fcast <- forecast(fit, h=168)
+fcast
+         Point Forecast      Lo 80     Hi 80         Lo 95     Hi 95
+4.000000       659.0645  566.71202  751.4169  517.82358229  800.3053
+4.005952       430.7339  325.02891  536.4388  269.07209741  592.3956
+4.011905       268.1259  157.28358  378.9682   98.60719504  437.6446
+4.017857       228.3024  116.08381  340.5210   56.67886523  399.9260
+4.023810       200.7340   88.25064  313.2174   28.70554423  372.7625
+4.029762       140.5758   28.04128  253.1103  -31.53088134  312.6824
+4.035714       196.1703   83.57555  308.7650   23.97150358  368.3690
+4.041667       282.6171  169.80545  395.4288  110.08657346  455.1476
+4.047619       446.6713  333.28115  560.0614  273.25604289  620.0865
+4.053571       479.9449  365.53618  594.3537  304.97184340  654.9180
+...
+```
+
+The output of `forecast` can be hard to decipher. You can plot the 
+forecasted values with the code below:
+
+```r
+# plot the values forecasted
+plot(fcast, include = 168, main="Taxicab Pickup Count in Times Square by Time", xlab="Date", ylab="Pickup Count", xaxt="n", col="red", fcol="blue")
+ticks <- seq(3, 5, 1/7)
+dates <- seq(as.Date("2016-01-15"), as.Date("2016-01-29"), by="days")
+dates <- format(dates, "%m-%d %H:%M")
+axis(1, at=ticks, labels=dates)
+legend('topleft', legend=c("Observed Value", "Predicted Value"), col=c("red", "blue"), lwd=c(2.5,2.5))
+
+# plot the observed values from the testing dataset
+count_rides_test$x <- seq(4, 4 + 239 * 1/168, 1/168)
+count_rides_test <- subset(count_rides_test, count_rides_test$one_hour < as.POSIXct("2016-01-29"))
+lines(count_rides_test$x, count_rides_test$count, col="red")
+```
+
+![alt text][rides_count_graph]
```

Comment:
Minor thing that's not mentioned in the readme (sorry!).  To properly show images, the html element needs a specific class associated with it, so you need to use an html <img> tag here.  Specifically `<img class="main-content__illustration" src="<your_source>" alt="<your_alt>">`